src/_C002.ino

```diff
@@ -148,14 +148,18 @@ bool CPlugin_002(byte function, struct EventStruct *event, String& string)
               }
 
               if (action.length() > 0) {
-                ExecuteCommand_plugin(x, VALUE_SOURCE_MQTT, action.c_str());
-
-                // trigger rulesprocessing
-                if (Settings.UseRules) {
-                  struct EventStruct TempEvent;
-                  TempEvent.TaskIndex = x;
-                  parseCommandString(&TempEvent, action);
-                  createRuleEvents(&TempEvent);
+                if (ExecuteCommand_internal(VALUE_SOURCE_MQTT, action.c_str())) {
+                } else {
+                  ExecuteCommand_plugin(x, VALUE_SOURCE_MQTT, action.c_str());
+                  // trigger rulesprocessing
+                  //SP: WHY WE NEED TO CALL rules FROM HERE?
```

Comment:
It is sending out an event which may trigger actions in a rules setup.
After all it is a command received from an external source, which can be an event to trigger some action.

Reply:
Understood. But on C005 (openHAB) there no such call and actions in the rules are triggered anyway.

Reply:
That's for sure something I don't like about the current situation.
Some controllers may require stuff to be done differently, but just the fact that we apparently need special knowledge about some plugins in the controllers is indication of a flaw in the design.
We clearly are missing something in the ESPEasy design and I am not entirely sure what.

For sure we do need some kind of "receiver" as the opposite of a controller we have now, but I am not entirely sure if that's what is missing for the event sending you pointed at here.

I do think that's actually part of the responsibility of the command handling.
So a controller may receive a command (let's assume it is also a task of a controller), but then it should hand it over to a central command handler.

My suggestion is to leave that part out of the scope of your PR as it may open a can of worms and Pandora's box at the same time.
I will add an issue for it.

Reply:
Added a new issue for it: https://github.com/letscontrolit/ESPEasy/issues/2865

Reply:
ok